python/tests/api/writer/test_whylabs.py

```diff
@@ -1,39 +1,181 @@
+from whylogs.api.whylabs.session.session_manager import (
+    SessionManager,
+    get_current_session,
+    init,
+)
+from whylogs.api.whylabs.session.session_types import SessionType
+from whylogs.api.whylabs.session.whylabs_client_cache import WhylabsClientCache
 from whylogs.api.writer import Writers
 from whylogs.api.writer.whylabs import WhyLabsWriter
 from whylogs.core.feature_weights import FeatureWeights
 from whylogs.core.schema import DatasetSchema
 from whylogs.core.segmentation_partition import segment_on_column
 
 logger = logging.getLogger(__name__)
+dirs = PlatformDirs("whylogs_tests", "whylogs")
+
+
+def _random_str(n: int) -> str:
+    return "".join(random.choices(string.ascii_letters + string.digits, k=n))
+
+
+class TestWhylabsWriterWithSession(object):
+    # So we don't delete our own configs while running tests
+    def setup_method(self) -> None:
+        WhylabsClientCache.reset()
+        SessionManager.reset()
+
+        os.environ[EnvVariableName.WHYLOGS_CONFIG_PATH.value] = f"{dirs.user_cache_dir}/whylogs_{_random_str(5)}.ini"
+
+        config = SessionConfig()
+        config.reset_config()
+
+    def teardown_method(self) -> None:
+        WhylabsClientCache.reset()
+        SessionManager.reset()
+        del os.environ[EnvVariableName.WHYLOGS_CONFIG_PATH.value]
+
+    def test_writer_throws_for_anon_sessions(self) -> None:
+        session = init()  # Default session is anonymous in this case (no config file content)
+        assert session.get_type() == SessionType.WHYLABS_ANONYMOUS
+
+        with pytest.raises(ValueError):
+            WhyLabsWriter()
+
+    def test_writer_works_for_anon_with_overrides(self) -> None:
+        key_id = "MPq7Hg002z"
+        org_id = "org-xxxxxx"
+        api_key = f"{key_id}.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx:{org_id}"
+
+        # You can use the writer if the session is anonymous, but you must provide the required args
+        session = init()  # Default session is anonymous in this case (no config file content)
+        assert session.get_type() == SessionType.WHYLABS_ANONYMOUS
+
+        # No error
+        writer = WhyLabsWriter(
+            api_key=api_key,
+            dataset_id="dataset_id",
+            org_id="org_id",
+        )
+
+        assert writer.key_id == api_key.split(".")[0]
+
+    def test_writer_uses_session_for_creds(self) -> None:
+        key_id = "MPq7Hg002z"
+        org_id = "org-xxxxxx"
+        dataset_id = "model-2"
+        api_key = f"{key_id}.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx:{org_id}"
+        session = init(whylabs_api_key=api_key, default_dataset_id=dataset_id)
+
+        assert session.get_type() == SessionType.WHYLABS
+
+        writer = WhyLabsWriter()
+        assert writer.key_id == key_id
+        assert writer._org_id == org_id
+        assert writer._dataset_id == dataset_id
+
+    def test_writer_uses_session_for_creds_implicitly(self) -> None:
```

Comment:
I tested some notebooks without modification, and now they are working as expected. The only issue now are the error messages - if I log 7 days worth of data with the following code:
```python
    profile = why.log(df).profile()
    profile.set_dataset_timestamp(timestamp)
    writer.write(profile.view())
```
I get one initial message:

`✅ Using session type: LOCAL. Profiles won't be uploaded or written anywhere automatically.`

And a series of messages (one for each day):

`Skipping uploading profiles to WhyLabs because this session is configured to be local only.`

I understand that this is related to the `log()` statement, but for the user this gives the impression that no data is being uploaded at all, and that there is something wrong.

Maybe we should rewrite the message so it is clearer that this is related only to the log statement? and maybe reducing the number of messages or changing the type of message?

Reply:
I updated the message to this

```
⚠️ Skipping automatic upload because the session type is LOCAL. Uploads have to be done manually.
```

And I made it only happen once as well. Now the output of the classification notebook looks like this

```
✅ Using session type: LOCAL. Profiles won't be uploaded or written anywhere automatically.
logging data for date 2023-08-10 00:00:00+00:00
⚠️ Skipping automatic upload because the session type is LOCAL. Uploads have to be done manually.
writing profiles to whylabs...
logging data for date 2023-08-11 00:00:00+00:00
writing profiles to whylabs...
logging data for date 2023-08-12 00:00:00+00:00
writing profiles to whylabs...
logging data for date 2023-08-13 00:00:00+00:00
writing profiles to whylabs...
logging data for date 2023-08-14 00:00:00+00:00
writing profiles to whylabs...
logging data for date 2023-08-15 00:00:00+00:00
writing profiles to whylabs...
```